components/cam/src/physics/cam/micro_p3.F90

```diff
@@ -896,222 +895,65 @@ SUBROUTINE p3_main(qc,nc,qr,nr,th_old,th,qv_old,qv,dt,qitot,qirim,nitot,birim,ss
+          call ice_self_collection(rho(i,k),rhofaci(i,k),&
+          f1pr03,eii,Eii_fact,qitot_incld(i,k),nitot_incld(i,k),&
+               nislf)
 
           !............................................................
           ! melting
-
-          ! need to add back accelerated melting due to collection of ice mass by rain (pracsw1)
-          ! note 'f1pr' values are normalized, so we need to multiply by N
-
-          if (qitot_incld(i,k).ge.qsmall .and. t(i,k).gt.zerodegc) then
-             qsat0 = 0.622_rtype*e0/(pres(i,k)-e0)
-             !  dum=cpw/xlf(i,k)*(t(i,k)-273.15)*(pracsw1+qcshd)
-             ! currently enhanced melting from collision is neglected
-             ! dum=cpw/xlf(i,k)*(t(i,k)-273.15)*(pracsw1)
-             dum = 0._rtype
-             ! qimlt=(f1pr05+f1pr14*sc**0.3333*(rhofaci(i,k)*rho(i,k)/mu)**0.5)* &
-             !       (t(i,k)-273.15)*2.*pi*kap/xlf(i,k)+dum
-             ! include RH dependence
-             qimlt = ((f1pr05+f1pr14*sc**thrd*(rhofaci(i,k)*rho(i,k)/mu)**0.5_rtype)*((t(i,k)-   &
-                  zerodegc)*kap-rho(i,k)*xxlv(i,k)*dv*(qsat0-qv(i,k)))*2._rtype*pi/xlf(i,k)+     &
-                  dum)*nitot_incld(i,k)
-             qimlt = max(qimlt,0.)
-             nimlt = qimlt*(nitot_incld(i,k)/qitot_incld(i,k))
-          endif
+          call ice_melting(rho(i,k),t(i,k),pres(i,k),rhofaci(i,k),&
+          f1pr05,f1pr14,xxlv(i,k),xlf(i,k),dv,sc,mu,kap,&
+          qv(i,k),qitot_incld(i,k),nitot_incld(i,k),&
+               qimlt,nimlt)
 
           !............................................................
           ! calculate wet growth
-
-          ! similar to Musil (1970), JAS
-          ! note 'f1pr' values are normalized, so we need to multiply by N
-
-          if (qitot_incld(i,k).ge.qsmall .and. qc_incld(i,k)+qr_incld(i,k).ge.1.e-6_rtype .and. t(i,k).lt.zerodegc) then
-
-             qsat0  = 0.622_rtype*e0/(pres(i,k)-e0)
-             qwgrth = ((f1pr05 + f1pr14*sc**thrd*(rhofaci(i,k)*rho(i,k)/mu)**0.5_rtype)*       &
-                  2._rtype*pi*(rho(i,k)*xxlv(i,k)*dv*(qsat0-qv(i,k))-(t(i,k)-zerodegc)*           &
-                  kap)/(xlf(i,k)+cpw*(t(i,k)-zerodegc)))*nitot_incld(i,k)
-             qwgrth = max(qwgrth,0._rtype)
-             !calculate shedding for wet growth
-             dum    = max(0._rtype,(qccol+qrcol)-qwgrth)
-             if (dum.ge.1.e-10_rtype) then
-                nrshdr = nrshdr + dum*1.923e+6_rtype   ! 1/5.2e-7, 5.2e-7 is the mass of a 1 mm raindrop
-                if ((qccol+qrcol).ge.1.e-10_rtype) then
-                   dum1  = 1._rtype/(qccol+qrcol)
-                   qcshd = qcshd + dum*qccol*dum1
-                   qccol = qccol - dum*qccol*dum1
-                   qrcol = qrcol - dum*qrcol*dum1
-                endif
-                ! densify due to wet growth
-                log_wetgrowth = .true.
-             endif
-
-          endif
-
-
+          call wet_growth(rho(i,k),t(i,k),pres(i,k),rhofaci(i,k),&
```

Comment:
I like having the names be of the form <from>_<to>_<process>, so can we name this cldliq_wetgrowth or ice_cldliq_wetgrowth (I can't keep straight whether this actually changes ice or not)...